src/WakuRlnV2.sol

```diff
@@ -59,27 +60,63 @@ contract WakuRlnV2 {
         if (!isValidCommitment(idCommitment)) revert InvalidIdCommitment(idCommitment);
         _;
     }
 
-    modifier onlyValidUserMessageLimit(uint32 messageLimit) {
-        if (messageLimit > MAX_MESSAGE_LIMIT) revert InvalidUserMessageLimit(messageLimit);
-        if (messageLimit == 0) revert InvalidUserMessageLimit(messageLimit);
+    /// @notice the modifier to check if the userMessageLimit is valid
+    /// @param userMessageLimit The user message limit
+    modifier onlyValidUserMessageLimit(uint32 userMessageLimit) {
+        if (!isValidUserMessageLimit(userMessageLimit)) revert InvalidUserMessageLimit(userMessageLimit);
         _;
     }
 
+    /// @notice the constructor of the contract
     constructor(uint32 maxMessageLimit) {
         MAX_MESSAGE_LIMIT = maxMessageLimit;
         SET_SIZE = uint32(1 << DEPTH);
         deployedBlockNumber = uint32(block.number);
         LazyIMT.init(imtData, DEPTH);
     }
 
-    function memberExists(uint256 idCommitment) public view returns (bool) {
+    /// @notice Checks if a commitment is valid
+    /// @param idCommitment The idCommitment of the member
+    /// @return true if the commitment is valid, false otherwise
+    function isValidCommitment(uint256 idCommitment) public pure returns (bool) {
+        return idCommitment != 0 && idCommitment < Q;
+    }
+
+    /// @notice Checks if a user message limit is valid
+    /// @param userMessageLimit The user message limit
+    /// @return true if the user message limit is valid, false otherwise
+    function isValidUserMessageLimit(uint32 userMessageLimit) public view returns (bool) {
+        return userMessageLimit > 0 && userMessageLimit <= MAX_MESSAGE_LIMIT;
+    }
+
+    /// @notice Returns the rateCommitment of a member
+    /// @param index The index of the member
+    /// @return The rateCommitment of the member
+    function indexToCommitment(uint32 index) internal view returns (uint256) {
+        return imtData.elements[LazyIMT.indexForElement(0, index)];
+    }
+
+    /// @notice Returns the metadata of a member
+    /// @param idCommitment The idCommitment of the member
+    /// @return The metadata of the member (userMessageLimit, index, rateCommitment)
+    function idCommitmentToMetadata(uint256 idCommitment) public view returns (uint32, uint32, uint256) {
         MembershipInfo memory member = memberInfo[idCommitment];
-        return member.userMessageLimit > 0 && member.index >= 0;
+        return (member.userMessageLimit, member.index, indexToCommitment(member.index));
+    }
+
+    /// @notice Checks if a member exists
+    /// @param idCommitment The idCommitment of the member
+    /// @return true if the member exists, false otherwise
+    function memberExists(uint256 idCommitment) public view returns (bool) {
+        (uint32 userMessageLimit, uint32 index, uint256 rateCommitment) = idCommitmentToMetadata(idCommitment);
+        return userMessageLimit > 0 && index >= 0 && rateCommitment != 0;
```

Comment:
Is there any case in which `index < 0` ?

Reply:
nope, it is not possible to have a negative index

Reply:
Then
```solidity
return userMessageLimit > 0 && rateCommitment != 0;
```

would be enough right?

Reply:
hmm, we could just do `rateCommitment != 0` actually, fixing.

Reply:
addressed in [f753ef5](https://github.com/waku-org/waku-rlnv2-contract/pull/5/commits/f753ef5c43193cb03d96ddc54aab972c2b80b928)

in doing so, discovered a bug where it would actually return the rateCommitment of the 0th index, because of the `indexToCommitment` call (which would default to zero for a member that does not exist). Here, we also add the check if the userMessageLimit is 0, we can infer that the member does not exist, and therefore, return 0 for the rateCommitment instead.

Reply:
Nice! Cool that you caught that one